### PR TITLE
Add a blurb on the x86 calling convention to clr-abi.md

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Add some information about the x86 calling convention based on my learnings from #39294

### CODE REVIEW

File: docs/design/coreclr/botr/clr-abi.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # CLR ABI
 
-This document describes the .NET Common Language Runtime (CLR) software conventions (or ABI, "Application Binary Interface"). It focusses on the ABI for the x64 (aka, AMD64), ARM (aka, ARM32 or Thumb-2), and ARM64 processor architectures. Documentation for the x86 ABI is somewhat scant.
+This document describes the .NET Common Language Runtime (CLR) software conventions (or ABI, "Application Binary Interface"). It focusses on the ABI for the x64 (aka, AMD64), ARM (aka, ARM32 or Thumb-2), and ARM64 processor architectures. Documentation for the x86 ABI is somewhat scant, but information on the basics of the calling convention are included at the bottom of this document.
```
Comment (sandreenko) [inline docs/design/coreclr/botr/clr-abi.md:3]:
```suggestion
This document describes the .NET Common Language Runtime (CLR) software conventions (or ABI, "Application Binary Interface"). It focusses on the ABI for the x64 (aka, AMD64), ARM (aka, ARM32 or Thumb-2), and ARM64 processor architectures. Documentation for the x86 ABI is somewhat scant, but information on the basics of the calling convention is included at the bottom of this document.
```

File: docs/design/coreclr/botr/clr-abi.md [hunk 2]
```diff
@@ -705,3 +705,7 @@ The general rules outlined in the [System V x86_64 ABI](https://software.intel.c
 | %xmm2-%xmm7  | used to pass floating point arguments   | No                |
 | %xmm8-%xmm15 | temporary registers                     | No                |
 ```
+
+# x86 Calling convention specifics
+
+The managed x86 calling convention uses a variant of the Windows x86 "fastcall" calling convention. The first two register-eligible arguments are passed in ecx and edx respectively. 1, 2, or 4 byte integer primitives and structs that trivially wrap a 4-byte primitive integer are eligible for registers. Structs that do not trivially wrap a 4-byte primitive integer or a register-eligible struct are passed on the stack. The this argument and the hidden arguments mentioned earlier in this document are also eligible for registers. All non-register eligible arguments are pushed on the stack in argument order. This is in contrast to the unmanaged calling conventions, cdecl, stdcall, thiscall, and fastcall, which push the arguments in reverse order onto the stack.
```
Comment (sandreenko) [inline docs/design/coreclr/botr/clr-abi.md:711]: Could you please start with information that x86 windows managed and native calling conventions are different in contrast with other platforms (including x86 Unix)?

Comment (jkoritzinsky) [reply docs/design/coreclr/botr/clr-abi.md:711]: The x86 managed calling convention for Unix is not the same as the native Unix x86 calling convention. They also have the differences I've listed. I'll add a blurb calling out the difference for x86 vs other architectures.

File: docs/design/coreclr/botr/clr-abi.md (hunk 1 above)
Comment (jkotas) [inline docs/design/coreclr/botr/clr-abi.md]: We had the x86 calling convention written down in the augmented version of the CLI spec.

Download https://download.microsoft.com/download/D/C/1/DC1B219F-3B11-4A05-9DA3-2D0F98B20917/Partition%20II%20Metadata.doc and look for "Managed native calling conventions (x86)"

Comment (jkoritzinsky) [reply docs/design/coreclr/botr/clr-abi.md]: I'll port the text from there over and update it.

**Review by sandreenko — APPROVED** [review]
LGTM, thanks for adding this.

**Pull request merged**